src/main/java/codeu/model/data/Bot.java

```diff
@@ -1,49 +1,9 @@
 package codeu.model.data;
 
-import codeu.model.store.basic.UserStore;
-import java.time.Instant;
-import java.util.UUID;
-import org.mindrot.jbcrypt.BCrypt;
+/** Interface for Bot implementations. */
+public interface Bot {
```

Comment:
Thinking a little more about it, how do you think about making the Bot interface contain a `mentionKeyword` field that each Bot implementation populates, then the `registerBot` can just be called with the Bot instance?
The reason I'm suggesting this is to also make the Bot aware of its own mention keyword, which seems reasonable to me. You can imagine the About Me section of the Bot for example listing its mention keyword.
It also kind of cruel to make Bots that don't know their own name. :)

Reply:
Haha, yes I'll add that, thanks for pointing this out. Does this also mean that using a set instead of a map would be more appropriate?

Reply:
Also, if we include that field I'm guessing an abstract class would be used over an interface, right?

Reply:
We'd still need a map because the registry needs to look up bots by the mention key. Sets are only useful to check if a certain bot is registered or not, i.e. look up by instance.
You can still use an interface that contain a string field. I know you can do that in other languages at least, but I suspect Java supports that too.

Reply:
Ah, that seems like a limitation in Java. How about a `getName` method instead then?

Reply:
Added 👍